starter-core/src/main/java/io/micronaut/starter/feature/database/HibernateGorm.java

```diff
@@ -70,12 +70,16 @@ public void processSelectedFeatures(FeatureContext featureContext) {
     @Override
     public void apply(GeneratorContext generatorContext) {
         Map<String, Object> config = generatorContext.getConfiguration();
-        config.put("dataSource.url", "jdbc:h2:mem:devDb;MVCC=TRUE;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE");
+        DatabaseDriverFeature dbFeature = generatorContext.getFeature(DatabaseDriverFeature.class);
+        if (dbFeature == null) {
+            dbFeature = defaultDbFeature;
+        }
```

Comment:
The `getFeature` method should return an optional, with a `getRequiredFeature` that returns the feature and throws an exception if it doesn't exist. In this scenario we should expect that the feature exists.

Reply:
Yeah, I had something close to that at first. The difficulty I ran into was that `FeatureSpec` calls `applyFeatures` on every feature without first calling `processSelectedFeatures`, so the database features aren't present and it would result in an exception.

Do you have any thoughts on how to handle that? Should I try to add a database feature always in `FeatureSpec`? That might not make sense for some features, but it might be messy to try to detect which features are the ones that require it and then add it. Or should I just leave the `null` handling here?

Reply:
@benmccann The line in the test should probably be `getFeatures([feature.name]).getFeatures()` then

Reply:
Sorry, I'm not following. Are you suggesting I go ahead and add `H2` to the `generatorContext` of for every feature in the test whether or not it's DB related?

Reply:
@jameskleeh thanks for the review! I'd love to close this out. Any advice on moving forward?